lib/Mojo/mysql/PubSub.pm

```diff
@@ -100,14 +131,57 @@ sub _wait_db {
     my $sync_db = $self->mysql->db;
     return (delete $self->{wait_db}, $self->_wait_db($sync_db)) if $err;
     $res->finish;
-    $db->query('select sleep(600)', $cb);
-    $sync_db->query('update mojo_pubsub_subscribe set ts = current_timestamp where pid = ?', $db->pid);
+    _query_with_retry(
+      $db,
+      'select sleep(600)',
+      $cb
+    );
+    _query_with_retry(
+      $sync_db,
+      'update mojo_pubsub_subscribe set ts = current_timestamp where pid = ?',
+      $db->pid
+    );
     $self->_notifications($self->mysql->db);
   };
 
   warn qq|[PubSub] (@{[$wait_db->pid]}) reconnect\n| if DEBUG;
   $self->emit(reconnect => $wait_db);
-  return $wait_db->query('select sleep(600)', $cb);
+  return _query_with_retry(
+    $wait_db,
+    'select sleep(600)',
+    $cb
+  );
+}
+
+sub _query_with_retry {
+  my ($db, @query_args) = @_;
+
+  my $wantarray = wantarray;
+  my @results;
+
+  my $remaining_attempts = 2;
```

Comment:
Should this be configurable? Maybe add an attribute or environment variable? Might be useful for someone to set it to 0? Example:

```
my $remaining_attempts = $self->number_of_retries + 1;
```

...though I'm not sure if `number_of_retries()` is a good attribute name.

Reply:
Good idea, I will try making it an attribute.  Maybe `max_allowed_retries` or `max_allowed_query_retries`?

Reply:
I think max_allowed_retries() or just adding an environment variable might be an alternative as well:

use constant RETRIES => $ENV{MOJO_MYSQL_PUBSUB_RETRIES} || 2;